src/Shared/Debugging/DebugUtils.cs

```diff
@@ -44,6 +59,32 @@ static DebugUtils()
             DebugPath = debugDirectory;
         }
 
+        private static string ProbeWriteability(string path)
```

Comment:
I don't like the naming here. Something like `CanWriteToDirectory` or similar is more intuitive.

See my comment above, but this function can catch its own `ArgumentException` and return null in that case.

We can consider placing this in `FileUtilities` if we can generalize this.

Reply:
I think the general form should just return true or false for can write/can't write. Should be doable.